packages/components/src/components/tooltip/tooltip.js

```diff
@@ -151,71 +177,183 @@ class Tooltip extends mixin(
+        }
+
+        if (this._handleClickListener) {
+          this._handleClickListener = this.unmanage(
+            this._handleClickListener
+          ).release();
+        }
+
+        if (this._handleKeydownListener) {
+          this._handleKeydownListener = this.unmanage(
+            this._handleKeydownListener
+          ).release();
+        }
+
+        if (state === 'shown') {
+          this._hookCloseActions(this.element);
+          const focusableNode = this.tooltip.element.querySelector(
+            settings.selectorTabbable
+          );
+          if (focusableNode) {
+            focusableNode.focus();
+          } else {
+            this.tooltip.element.setAttribute('tabindex', '0');
+            this.tooltip.element.focus();
+          }
+
+          const hasFocusin =
+            'onfocusin' in this.element.ownerDocument.defaultView;
+          const focusinEventName = hasFocusin ? 'focusin' : 'focus';
+          this._handleFocusinListener = this.manage(
+            on(
+              this.element.ownerDocument,
+              focusinEventName,
+              this._handleFocusin,
+              !hasFocusin
+            )
+          );
+        } else {
+          this.element.focus();
+        }
+
+        // @question Do I need to pass any parameters to callback?
+        if (typeof callback === 'function') callback();
+      }
     );
   }
 
   /**
-   * Attaches event handlers to show/hide the tooltip.
+   * Attaches event handlers to show the tooltip.
    * @param {Element} element The element to attach the events to.
    * @private
    */
   _hookOn(element) {
-    const hasFocusin = 'onfocusin' in window;
-    const focusinEventName = hasFocusin ? 'focusin' : 'focus';
-    [focusinEventName, 'blur', 'touchleave', 'touchcancel'].forEach(name => {
+    this.manage(
+      on(
+        element,
+        'click',
+        event => {
+          const { relatedTarget, type } = event;
+          const hadContextMenu = this._hasContextMenu;
+          this._hasContextMenu = type === 'contextmenu';
+          this._debouncedHandleClick({
+            relatedTarget,
+            type,
+            hadContextMenu,
+            details: getLaunchingDetails(event),
+          });
+        },
+        false
+      )
+    );
+
+    if (this.element.tagName.toLowerCase() !== 'button') {
       this.manage(
         on(
-          element,
-          name,
+          this.element,
+          // Does Carbon prefer keydown or keyup?
+          'keydown',
           event => {
-            const { relatedTarget, type } = event;
-            const hadContextMenu = this._hasContextMenu;
-            this._hasContextMenu = type === 'contextmenu';
-            this._debouncedHandleClick({
-              relatedTarget,
-              type: type === 'focusin' ? 'focus' : type,
-              hadContextMenu,
-              details: getLaunchingDetails(event),
-            });
+            const { relatedTarget, type, which } = event;
```

Comment:
Given the this event handler implementation is used in more than one place (or at least similar), would you want to create a private method for that?